src/classes/HiFiMixerSession.ts

```diff
@@ -1231,6 +1232,25 @@ export class HiFiMixerSession {
         this.mixerInfo = {
             "connected": false,
         };
+    }
+
+    /** 
+     * Clears all cached peer data and submits all hashedVisitIDs as "deleted"
+     */
+    private _clearPeerData(): void {
+        if (this.onUsersDisconnected && this._mixerPeerKeyToStateCacheDict.length > 0) {
+            let allDeletedUserData: Array<ReceivedHiFiAudioAPIData> = [];
+            for (const peerData of this._mixerPeerKeyToStateCacheDict) {
+                let deletedUserData = new ReceivedHiFiAudioAPIData({
+                    hashedVisitID: peerData.hashedVisitID,
+                });
+                if (peerData.providedUserID) {
+                    deletedUserData.providedUserID = peerData.providedUserID;
+                }
+                allDeletedUserData.push(deletedUserData);
+            }
+            this.onUsersDisconnected(allDeletedUserData);
+        }
         this._mixerPeerKeyToStateCacheDict = {};
     }
 }
```

Comment:
Note: we aren't just clearing the peer cache here.  If **onUserDisconnect** callback exists we also assemble **allDeletedUserData** and feed it to **onUserDisconnect**.  This tells the Application to remove any stale peer data it happens to be using.